### PR TITLE
Separating create iteration modal from main iteration component.

### PR DESCRIPTION
**Pull request by vikram-raj**
fix: #808

### CODE REVIEW

**Comment by nimishamukherjee** [comment]
This is a specific Iteration Modal and not a generic one?

File: src/app/iteration/iteration-modal/iteration-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+@Component({
+  selector: 'fab-planner-iteration-modal',
+  templateUrl: './iteration-modal.component.html',
+  styleUrls: ['./iteration-modal.component.scss']
+})
+export class FabPlannerIterationModalComponent {
+  @ViewChild('createUpdateIterationDialog') createUpdateIterationDialog: any;
+
+  openModal() {
```
Comment (debloper) [inline src/app/iteration/iteration-modal/iteration-modal.component.ts]: I'd prefer `createIteration()` as a name.

File: src/app/iteration/iteration-modal/iteration-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<modal #createUpdateIterationDialog title="Create Iteration">
+  <modal-content>
+    <div class="col-md-12 col-sm-12 col-xs-12 createProjectBlock">
+      <div class="form-group">
+        <div class="search-pf-input-group">
+          <label for="iteration-name">Name <span class="field-required-icon">*</span></label>
+          <input id="iteration-name" type="text" class="form-control" placeholder="Enter a unique iteration name">
```
Comment (debloper) [inline src/app/iteration/iteration-modal/iteration-modal.component.html]: Drop `a unique` in here.

File: src/app/iteration/iteration-modal/iteration-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          <span class="help-block">Iteration names must be unique within a project</span>
```
Comment (debloper) [inline src/app/iteration/iteration-modal/iteration-modal.component.html]: Emphasize `must be unique` in here.

File: src/app/iteration/iteration.component.html [hunk 4]
```diff
@@ -187,30 +187,10 @@
     </ul>
   </div>
 </div>
-<div *ngIf="loggedIn" class="create-iteration-bar pointer" (click)="createIterationDialog.open()">
+<div *ngIf="loggedIn" class="create-iteration-bar pointer" (click)="modal.openModal()">
```
Comment (debloper) [inline src/app/iteration/iteration.component.html]: `modal.createIteration()`

**Comment by debloper** [comment]
Made some inline comments.
Also, after this, there has to be a big `s/alm/fab/` in our codebase.

**Comment by vikram-raj** [comment]
@nimishamukherjee yes it is specific iteration modal.

File: src/app/iteration/iteration-modal/iteration-modal.component.ts (hunk 1 above)
Comment (sanbornsen) [inline src/app/iteration/iteration-modal/iteration-modal.component.ts]: It actually for create and update so I'd go for openCreateUpdateModal()

Comment (vikram-raj) [inline src/app/iteration/iteration-modal/iteration-modal.component.ts]: As it is for create and update iteration modal. I rename it to openCreateUpdateModal().

**Comment by nimishamukherjee** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Comment by sanbornsen** [comment]
[test]

**Pull request merged**